Makefile

```diff
@@ -82,10 +82,16 @@ install: all
 	  install -m 644 $(JS_FILES) "$(DESTDIR)$(htmldir)"/; \
 	fi
 
-validate:
-	@xmllint --noout --dtdvalid devbook.dtd $(XMLS)
+# Not all versions of xmllint support --quiet, so test for it first
+validate: devbook.rng
+	@opt=--quiet; xmllint --help 2>&1 | grep -q -- --quiet || opt=; \
+	xmllint --noout $${opt} --relaxng $< $(XMLS)
 	@echo "xmllint validation successful"
 
+%.rng: %.rnc
+	trang -I rnc -O rng $< $@
```

Comment:
Could you update `appendices/contributing/text.xml` to note this new dependency?

Reply:
Not sure if we should mention it. I went for the pragmatic approach to commit also `devbook.rng` to the repository, so trang won't normally be needed, neither for editing the XML sources nor for building the devmanual. Purists may object against having a generated file in the repository, but it is a small and machine-independent file, and it avoids app-text/trang as a dependency, which in turn depends on Java.

I very much trust that someone who will find themselves in the position to edit the RELAX NG schema will also know how to install trang. :smiley:

Reply:
Okay, good point. WFM!